hazelcast/src/main/java/com/hazelcast/query/impl/getters/JsonDataGetter.java

```diff
@@ -18,39 +18,47 @@
 
 import com.fasterxml.jackson.core.JsonFactory;
 import com.fasterxml.jackson.core.JsonParser;
+import com.hazelcast.internal.nio.BufferObjectDataInput;
+import com.hazelcast.internal.serialization.Data;
 import com.hazelcast.internal.serialization.InternalSerializationService;
-import com.hazelcast.internal.serialization.impl.DataInputNavigableJsonAdapter;
 import com.hazelcast.internal.serialization.impl.NavigableJsonInputAdapter;
-import com.hazelcast.internal.serialization.Data;
 
 import java.io.IOException;
 
 import static com.hazelcast.internal.serialization.impl.HeapData.HEAP_DATA_OVERHEAD;
 
 public final class JsonDataGetter extends AbstractJsonGetter {
 
-    private static final int UTF_CHARACTER_COUNT_FIELD_SIZE = 4;
+    private static final int UTF_BYTE_COUNT_FIELD_SIZE = 4;
 
-    private JsonFactory factory = new JsonFactory();
+    private final JsonFactory factory = new JsonFactory();
 
-    private InternalSerializationService ss;
+    private final InternalSerializationService ss;
 
-    JsonDataGetter(InternalSerializationService ss) {
+    public JsonDataGetter(InternalSerializationService ss) {
         super(null);
+
         this.ss = ss;
     }
 
     protected JsonParser createParser(Object obj) throws IOException {
         Data data = (Data) obj;
-        return factory.createParser(data.toByteArray(),
-                HEAP_DATA_OVERHEAD + UTF_CHARACTER_COUNT_FIELD_SIZE,
-                data.dataSize() - UTF_CHARACTER_COUNT_FIELD_SIZE);
+
+        return factory.createParser(
+            data.toByteArray(),
+            HEAP_DATA_OVERHEAD + UTF_BYTE_COUNT_FIELD_SIZE,
+            data.dataSize() - UTF_BYTE_COUNT_FIELD_SIZE
+        );
     }
 
     @Override
-    protected NavigableJsonInputAdapter annotate(Object object) {
```

Comment:
Minor: Since we know where to read from the byte-array in Data, an alternative questionable improvement can be like this with one less object creation: (but with the cost of doing same thing from another method)

```
    @Override
    protected NavigableJsonInputAdapter annotate(Object object) {
        Data data = (Data) object;

        assert data.isJson();

        int innerDataStartOffset = HeapData.DATA_OFFSET + 4;
        byte[] bytes = data.toByteArray();
        String result = new String(bytes, innerDataStartOffset, 
                           bytes.length - innerDataStartOffset, UTF_8);

        return new NavigableJsonInputAdapter(result, 0);
    }
```